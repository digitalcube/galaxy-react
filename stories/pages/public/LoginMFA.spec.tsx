import { render } from '@testing-library/react';
import React from 'react';
import { Default as LoginMFA, Inprogress, Failure } from './LoginMFA.stories';

describe('LoginMFA', () => {
  it('to match snapshot(default)', () => {
    const { asFragment } = render(<LoginMFA {...(LoginMFA.args as any)} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('to match snapshot (inprogress)', () => {
    const { asFragment } = render(<Inprogress {...(Inprogress.args as any)} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('to match snapshot (failure)', () => {
    const { asFragment } = render(<Failure {...(Failure.args as any)} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
