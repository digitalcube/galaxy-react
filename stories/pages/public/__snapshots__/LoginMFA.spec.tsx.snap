// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginMFA to match snapshot (failure) 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-column min-vh-100 shifter-dashboard-login"
  >
    <div
      class="mt-4 mb-4 flex-grow-1 d-flex align-items-center justify-content-center"
    >
      <div
        class="mx-auto d-flex flex-column flex-md-row align-items-center justify-content-center justify-content-md-between login-column"
      >
        <div
          class="login px-4 py-4 rounded d-flex flex-column justify-content-center mx-3 error"
        >
          <form
            novalidate=""
          >
            <div
              class="mb-4 pb-2 login-header"
            >
              <div
                class="mb-4 text-center login-icon"
              >
                <img
                  alt="Shifter"
                  height="46"
                  width="36"
                />
              </div>
              <h1
                class="mb-0 font-weight-bold text-center login-title"
              >
                Authentication
              </h1>
              <p
                class="text-center font-weight-bold"
              >
                Enter the verifcation code we sent to 
                <b>
                  ***-***-1234
                </b>
              </p>
            </div>
            <div
              class="alert alert-danger mb-4 p-0 border-0 d-flex c-alert"
              role="alert"
            >
              <div
                class="alert-icon px-1 d-flex align-items-center"
              >
                <i
                  aria-hidden="true"
                >
                  <svg
                    class="d-block"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                    />
                  </svg>
                </i>
              </div>
              <div
                class="alert-body px-3 py-2 flex-grow-1"
              >
                <p
                  class="mb-0"
                >
                  We found some errors with your login info. Please correct these issues to continue
                </p>
              </div>
            </div>
            <div
              class="form-row mx-0 verification-code"
            >
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="d-block invalid-feedback"
              >
                code is incorrect
              </div>
            </div>
            <button
              class="mt-4 btn-colored font-weight-bold btn btn-secondary btn-block disabled"
              disabled=""
              type="submit"
            >
              Log In
            </button>
            <div
              class="mt-4 py-3 font-weight-bold text-center back2login"
            >
              <a
                href="#"
                target="_blank"
              >
                Back to login
              </a>
            </div>
          </form>
        </div>
        <div
          class="login-information mt-4 mt-md-0 mx-3"
        >
          <div
            class="login-information-icon text-center"
          >
            <img
              alt=""
              height="141"
              width="153"
            />
          </div>
          <h2
            class="login-information-title mb-0 font-weight-bold text-center"
          >
            Faster Build Times
          </h2>
          <div
            class="login-information-description mt-4 text-center"
          >
            Customize Shifter settings for up to 70% faster build times.
          </div>
          <div
            class="login-information-anchor font-weight-bold text-center"
          >
            <a
              href="#"
              to="#"
            >
              Learn about Shifter Media CDN
            </a>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="text-center"
    >
      <p
        class="m-0 common-footer-copyright"
      >
        <small
          class="d-flex align-items-center justify-content-center"
        >
          Â©2021 Shifter | A Product by 
          <a
            href="https://www.digitalcube.jp/"
            rel="nofollow noopener"
            title="DigitalCube"
          >
            <img
              alt=""
              class="d-block"
              height="15"
              width="77"
            />
          </a>
        </small>
      </p>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`LoginMFA to match snapshot (inprogress) 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-column min-vh-100 shifter-dashboard-login"
  >
    <div
      class="mt-4 mb-4 flex-grow-1 d-flex align-items-center justify-content-center"
    >
      <div
        class="mx-auto d-flex flex-column flex-md-row align-items-center justify-content-center justify-content-md-between login-column"
      >
        <div
          class="login px-4 py-4 rounded d-flex flex-column justify-content-center mx-3 success"
        >
          <form
            novalidate=""
          >
            <div
              class="mb-4 pb-2 login-header"
            >
              <div
                class="mb-4 text-center login-icon"
              >
                <img
                  alt="Shifter"
                  height="46"
                  width="36"
                />
              </div>
              <h1
                class="mb-0 font-weight-bold text-center login-title"
              >
                Authentication
              </h1>
              <p
                class="text-center font-weight-bold"
              >
                Enter the verifcation code we sent to 
                <b>
                  ***-***-1234
                </b>
              </p>
            </div>
            <div
              class="form-row mx-0 verification-code"
            >
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <button
              class="mt-4 btn-colored font-weight-bold btn btn-secondary btn-block disabled"
              disabled=""
              type="submit"
            >
              Log In
            </button>
            <div
              class="mt-4 py-3 font-weight-bold text-center back2login"
            >
              <a
                href="#"
                target="_blank"
              >
                Back to login
              </a>
            </div>
          </form>
        </div>
        <div
          class="login-information mt-4 mt-md-0 mx-3"
        >
          <div
            class="login-information-icon text-center"
          >
            <img
              alt=""
              height="141"
              width="153"
            />
          </div>
          <h2
            class="login-information-title mb-0 font-weight-bold text-center"
          >
            Faster Build Times
          </h2>
          <div
            class="login-information-description mt-4 text-center"
          >
            Customize Shifter settings for up to 70% faster build times.
          </div>
          <div
            class="login-information-anchor font-weight-bold text-center"
          >
            <a
              href="#"
              to="#"
            >
              Learn about Shifter Media CDN
            </a>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="text-center"
    >
      <p
        class="m-0 common-footer-copyright"
      >
        <small
          class="d-flex align-items-center justify-content-center"
        >
          Â©2021 Shifter | A Product by 
          <a
            href="https://www.digitalcube.jp/"
            rel="nofollow noopener"
            title="DigitalCube"
          >
            <img
              alt=""
              class="d-block"
              height="15"
              width="77"
            />
          </a>
        </small>
      </p>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`LoginMFA to match snapshot(default) 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-column min-vh-100 shifter-dashboard-login"
  >
    <div
      class="mt-4 mb-4 flex-grow-1 d-flex align-items-center justify-content-center"
    >
      <div
        class="mx-auto d-flex flex-column flex-md-row align-items-center justify-content-center justify-content-md-between login-column"
      >
        <div
          class="login px-4 py-4 rounded d-flex flex-column justify-content-center mx-3"
        >
          <form
            novalidate=""
          >
            <div
              class="mb-4 pb-2 login-header"
            >
              <div
                class="mb-4 text-center login-icon"
              >
                <img
                  alt="Shifter"
                  height="46"
                  width="36"
                />
              </div>
              <h1
                class="mb-0 font-weight-bold text-center login-title"
              >
                Authentication
              </h1>
              <p
                class="text-center font-weight-bold"
              >
                Enter the 6 digit code from your authenticator app
              </p>
            </div>
            <div
              class="form-row mx-0 verification-code"
            >
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="col-2 px-0"
              >
                <input
                  class="form-control px-0 text-center"
                  inputmode="numeric"
                  maxlength="1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <button
              class="mt-4 btn-colored font-weight-bold btn btn-secondary btn-block disabled"
              disabled=""
              type="submit"
            >
              Log In
            </button>
            <div
              class="mt-4 py-3 font-weight-bold text-center back2login"
            >
              <a
                href="#"
                target="_blank"
              >
                Back to login
              </a>
            </div>
          </form>
        </div>
        <div
          class="login-information mt-4 mt-md-0 mx-3"
        >
          <div
            class="login-information-icon text-center"
          >
            <img
              alt=""
              height="141"
              width="153"
            />
          </div>
          <h2
            class="login-information-title mb-0 font-weight-bold text-center"
          >
            Faster Build Times
          </h2>
          <div
            class="login-information-description mt-4 text-center"
          >
            Customize Shifter settings for up to 70% faster build times.
          </div>
          <div
            class="login-information-anchor font-weight-bold text-center"
          >
            <a
              href="#"
              to="#"
            >
              Learn about Shifter Media CDN
            </a>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="text-center"
    >
      <p
        class="m-0 common-footer-copyright"
      >
        <small
          class="d-flex align-items-center justify-content-center"
        >
          Â©2021 Shifter | A Product by 
          <a
            href="https://www.digitalcube.jp/"
            rel="nofollow noopener"
            title="DigitalCube"
          >
            <img
              alt=""
              class="d-block"
              height="15"
              width="77"
            />
          </a>
        </small>
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
